### ISSUE TITLE
Allow recursive includes of a path under an excluded path

### ISSUE THREAD

**Issue by micheldorsey** [microsoft/vscode-cpptools#5549]
My project has all of the object code and include files from other projects in a directory named "build".  I exclude this directory via the Files: exclude setting.  If I need to set the include path for headers in a  subset of this build directory, it fails when using glob characters.

Example:
Files Exclude:  **/build/**
includePath:  "${workspaceFolder}/infrastructure/projects/service/build/generated/**"

The end result is no header files are included from the include path.  If I explicitly list every include path in the build directory, it will work - just not with **.

remote extensions
cpp-tools 0.28.1
vscode 1.45.1
Local extensions
remote - SSH 0.51.0

Running on windows 10 and using a remote project to Ubuntu 18.04

**Comment by sean-mcmanus** [comment]
"**" is the recursive includes feature, which I believe will check for files.exclude, so this sounds like it's "by design". Are you able to workaround that design to get it to do what you want? Or did you have an idea for an alternative design that wouldn't break other users/scenarios?

**Comment by micheldorsey** [comment]
If the the same code is used to populate the browser pane as the tagging feature that looks at the include paths, then I can see this as a difficult thing to workaround.  The code that builds the ipch files could look at the IncludePath setting without regard for the browser exclude settings.

**Comment by sean-mcmanus** [comment]
I think a setting would need to be added to alter the behavior.

**Comment by micheldorsey** [comment]
Hey @Sean-mcmanus,

I’m no longer working with C++ and vscode, so the exact scenario is getting a bit fuzzy in my head.  It seems to me that we were working with a very large code base and there were redundant include directories for lots of sub-projects of the code.  I think our build team wants each sub-project to have all of the code in its respective tree to build the target.  With this in mind, we needed to exclude all but one of these redundant include directories, but still include just one of them so all of the sub-project would build.

An example would be that we had the boost include tree (huge) as part of each sub-project tree.  Including all of the sub-projects into vscode was necessary in order to see all of the code that we wanted to edit, but then vscode would pull in many copies of boost, so then we excluded the boost directory using a wildcard.  That part worked, but then we needed to add back into the include path one of the boost directories in order for the tagging to work.

Hope that helps!  Keep up the great work on the editor.

Thanks!
Michel
